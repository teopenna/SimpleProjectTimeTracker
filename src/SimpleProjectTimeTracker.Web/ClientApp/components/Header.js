import React, { Component } from "react";
import { Link, Redirect } from 'react-router-dom';
import { RouteComponentProps } from 'react-router';

export class Header extends Component {
    render() {
        return (
            <nav className="navbar navbar-expand-lg navbar-light bg-light">
                <a className="navbar-brand" href="#">Simple Project Time Tracker</a>
                <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
                    <span className="navbar-toggler-icon"></span>
                </button>

                <div className="collapse navbar-collapse" id="navbarSupportedContent">
                    <ul className="navbar-nav mr-auto">
                        <li className="nav-item active">
                            <a className="nav-link" href="#">Home <span className="sr-only">(current)</span></a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="/timeregistrations">Time Registrations</a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="/projects">Projects</a>
                        </li>
                        <li className="nav-item">
                            <a className="nav-link" href="/invoices">Invoices</a>
                        </li>
                    </ul>
                </div>
            </nav>
        );
    }
}

Header.displayName = 'Header';