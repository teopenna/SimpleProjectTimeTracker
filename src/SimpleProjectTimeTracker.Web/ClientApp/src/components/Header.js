import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

class Header extends Component {
    render() {
        return (
            <nav className="navbar navbar-expand-lg navbar-light bg-light">
                <div className="container">
                    <NavLink className="navbar-brand" to="/">Simple Project Time Tracker</NavLink>
                    <div className="collapse navbar-collapse" id="navbarSupportedContent">
                        <ul className="navbar-nav mr-auto">
                            <li className="nav-item">
                                <NavLink className="nav-link" activeClassName="active" to="/timeregistrations">Time Registrations</NavLink>
                            </li>
                            <li className="nav-item">
                                <NavLink className="nav-link" activeClassName="active" to="/projects">Projects</NavLink>
                            </li>
                            <li className="nav-item">
                                <NavLink className="nav-link" activeClassName="active" to="/invoices">Invoices</NavLink>
                            </li>
                        </ul>
                    </div>
                </div>
            </nav>
        );
    }
}

export default Header;