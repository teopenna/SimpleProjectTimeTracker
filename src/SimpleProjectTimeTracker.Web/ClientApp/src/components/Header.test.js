import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

const setup = () => {
    const wrapper = shallow(<Header />);

    return {
        wrapper,
    }
};

describe('Header Component', () => {
    it('should render without crashing', () => {
        const { wrapper } = setup();
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should render three menu items', () => {
        const { wrapper } = setup();
        const menuItems = wrapper.find('li.nav-item');
        expect(menuItems.length).toBe(3);
    });
});