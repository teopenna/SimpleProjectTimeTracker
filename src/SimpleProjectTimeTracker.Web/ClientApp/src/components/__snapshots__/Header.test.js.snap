// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders without crashing 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="navbar-brand"
      to="/"
    >
      Simple Project Time Tracker
    </NavLink>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/timeregistrations"
          >
            Time Registrations
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/projects"
          >
            Projects
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/invoices"
          >
            Invoices
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Header Component should render without crashing 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="navbar-brand"
      to="/"
    >
      Simple Project Time Tracker
    </NavLink>
    <div
      className="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <ul
        className="navbar-nav mr-auto"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/timeregistrations"
          >
            Time Registrations
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/projects"
          >
            Projects
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/invoices"
          >
            Invoices
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
